txscript/data/script_tests.json

```diff
@@ -42,21 +42,516 @@
+["0", "IF ELSE 1 ENDIF", "NONE", "OK"],
+["1 1", "IF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "OK"],
+["1 0", "IF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "OK"],
+["1 1", "IF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "OK"],
+["0 0", "IF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "OK"],
+["0", "IF 0 ELSE 1 ELSE 0 ENDIF", "NONE", "OK", "Multiple ELSE's are valid and executed inverts on each ELSE encountered"],
+["1", "IF 1 ELSE 0 ELSE ENDIF", "NONE", "OK"],
+["1", "IF ELSE 0 ELSE 1 ENDIF", "NONE", "OK"],
+["1", "IF 1 ELSE 0 ELSE 1 ENDIF ADD 2 EQUAL", "NONE", "OK"],
+["1 2 0 IF AND ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["1 2 0 IF OR ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["1 2 0 IF XOR ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 0 IF 2MUL ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 0 IF 2DIV ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 2 0 IF MUL ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 2 0 IF DIV ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 2 0 IF MOD ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 2 0 IF LSHIFT ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["2 2 0 IF RSHIFT ELSE 1 ENDIF", "NOP", "NONE", "OK"],
+["'' 1", "IF SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ELSE ELSE SHA1 ENDIF 0x14 0x68ca4fec736264c13b859bac43d5173df6871682 EQUAL", "NONE", "OK"],
+["0", "IF 1 IF RETURN ELSE RETURN ELSE RETURN ENDIF ELSE 1 IF 1 ELSE RETURN ELSE 1 ENDIF ELSE RETURN ENDIF ADD 2 EQUAL", "NONE", "OK", "Nested ELSE ELSE"],
+["1 0x01 0x80", "IF 0 ENDIF", "NONE", "OK", "negative 0 is false"],
+["1", "IF 0x50 ENDIF 1", "NONE", "ERR_RESERVED_OPCODE", "0x50 is reserved"],
+["0", "DUP IF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0", "IF 1 ENDIF", "NONE", "ERR_EMPTY_STACK"],
+["0", "DUP IF ELSE ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0", "IF 1 ELSE ENDIF", "NONE", "ERR_EMPTY_STACK"],
+["1", "IF ELSE ENDIF ELSE", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "IF ELSE ENDIF ELSE ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "IF ENDIF ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "IF ELSE ELSE ENDIF ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["0 1", "IF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0 0", "IF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["1 0", "IF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0 1", "IF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["1", "IF 1", "NONE", "ERR_UNBALANCED_CONDITIONAL", "IF without ENDIF"],
+["1 IF 1", "ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL", "IFs don't carry over"],
+["1 IF", "1 ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL", "IF/ENDIF can't span scriptSig/scriptPubKey"],
+["1 IF 0 ENDIF", "1 ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["NOP", "IF 1 ENDIF", "NONE", "ERR_INVALID_STACK_OPERATION"],
```

Comment:
Since this test was added in this PR it needs a comment for what is being tested.